Lib/test/test_concurrent_futures.py

```diff
@@ -178,6 +178,20 @@ def test_thread_names_default(self):
             self.assertRegex(t.name, r'^.*ThreadPoolExecutor.*_[0-4]$')
             t.join()
 
+    def test_default_max_queue_size(self):
```

Comment:
I would add a higher-level test, i.e. create an executor with a non-zero queue size, submit a number of tasks larger than the queue size, and check that they all get executed in the end.